### PR TITLE
Feature/forget password

### LINKED ISSUE #15: Create forget password route

### PR DESCRIPTION
**Pull request by Rendy752**
Reset token expiration time 5 minute

### CODE REVIEW

**Comment by fanesz** [comment]
create separated branch from this branch #33

File: src/modules/auth/controllers/auth.controller.ts [hunk 1]
```diff
@@ -123,6 +129,94 @@ class AuthController extends BaseController {
...
+        }
+
+        const user = await this.userSubservice.findOne({ _id: userId });
+
+        if (!user) {
+          return this.handleError(res, {
+            statusCode: StatusNotFound,
+            message: "User not found",
+          });
+        }
```
Comment (fanesz) [inline src/modules/auth/controllers/auth.controller.ts]:
```suggestion
        const user = await this.userSubservice.findOne({ _id: userId });
        if (!user) {
          return this.handleError(res, {
            statusCode: StatusNotFound,
            message: "User not found",
          });
        }
```

Comment (fanesz) [reply src/modules/auth/controllers/auth.controller.ts]: no extra space to tell that this if check is error handling from previous line action

File: src/modules/auth/controllers/auth.controller.ts [hunk 2]
```diff
@@ -123,6 +129,94 @@ class AuthController extends BaseController {
...
+        const hashedPassword = await bcrypt.hash(newPassword as string, 10);
+
+        const updatedUser = await this.userSubservice.update(
+          { _id: userId },
+          { password: hashedPassword },
+        );
+
+        if (this.isServiceError(res, updatedUser)) {
+          return;
+        }
```
Comment (fanesz) [inline src/modules/auth/controllers/auth.controller.ts]:
```suggestion
        const updatedUser = await this.userSubservice.update(
          { _id: userId },
          { password: hashedPassword },
        );
        if (this.isServiceError(res, updatedUser)) {
          return;
        }
```

File: src/modules/auth/controllers/auth.controller.ts [hunk 3]
```diff
@@ -123,6 +129,94 @@ class AuthController extends BaseController {
...
+
+        await this.redisClient.del(`forget-password:${resetToken}`);
```
Comment (fanesz) [inline src/modules/auth/controllers/auth.controller.ts]:
```suggestion
        this.redisClient.del(`forget-password:${resetToken}`);
```
doesn't need to be await

**Review by fanesz — CHANGES_REQUESTED** [review]
Harusnyo sih ikutin kayak uvce, ado pengecekan di middleware untuk jwt reset pass. tapi dak usalah, low priority ini

**Comment by Rendy752** [comment]
okay

**Review by fanesz — APPROVED** [review]

**Pull request merged**